### PR TITLE
Improve performance of upgrade script

### PR DESCRIPTION
**Pull request by max-schaefer**
I noticed that this script was taking a very long time to upgrade a database of `cockroach` downloaded from LGTM.com, probably due to a bad join ordering. This seems to fix it.

### CODE REVIEW

File: upgrades/ee5c327face2866a7b3b12dcce5c291be52ebf52/upgrade_comment_groups.ql [hunk 1]
```diff
@@ -33,12 +33,15 @@ Location getLocation(CommentGroup cg) {
...
 from CommentGroup cg, File f, int idx
 where
-  f = getLocation(cg).getFile() and
-  rank[idx + 1](CommentGroup rankedcg |
-    getLocation(rankedcg).getFile() = f
-  |
-    rankedcg order by getLocation(rankedcg).getStartLine()
-  ) = cg
+  hasLocation(cg, f, _) and
+  cg = rank[idx + 1](CommentGroup cg2, int l | hasLocation(cg2, f, l) | cg2 order by l)
```
Comment (owen-mc) [inline upgrades/ee5c327face2866a7b3b12dcce5c291be52ebf52/upgrade_comment_groups.ql:46]: I'm not a big fan of `l` as an identifier. It looks very similar to `1` in some fonts.

Comment (max-schaefer) [reply upgrades/ee5c327face2866a7b3b12dcce5c291be52ebf52/upgrade_comment_groups.ql]: Fair enough; renamed to `line`. I've also renamed `f` to `file` at the same time for consistency.

File: upgrades/ee5c327face2866a7b3b12dcce5c291be52ebf52/upgrade_comment_groups.ql [hunk 2]
```diff
@@ -33,12 +33,15 @@ Location getLocation(CommentGroup cg) {
   not exists(cg.getLocation()) and result = cg.getComment(0).getLocation()
 }
 
-from CommentGroup cg, File f, int idx
+predicate hasLocation(CommentGroup cg, File f, int startLine) {
```
Comment (smowton) [inline upgrades/ee5c327face2866a7b3b12dcce5c291be52ebf52/upgrade_comment_groups.ql:36]: Nitpick: does the speedup depend on this not being inlined, which in turn relies on the number of calls being >= 2? If so suggest adding a pragma even if the current optimiser doesn't need it, to be robust against future strategy changes.

Comment (max-schaefer) [reply upgrades/ee5c327face2866a7b3b12dcce5c291be52ebf52/upgrade_comment_groups.ql:36]: Good idea; done.

**Pull request merged**